mav_trajectory_generation/src/segment.cpp

```diff
@@ -196,4 +196,59 @@ bool Segment::selectMinMaxMagnitudeFromCandidates(
   return true;
 }
 
+bool Segment::getSegmentWithSingleDimension(int dimension,
+                                            Segment* new_segment) const {
+  if (dimension < 0 || dimension >= D_) {
+    LOG(WARNING)
+        << "You shan't ask for a dimension that does not exist in the segment.";
+    return false;
+  }
+
+  *new_segment = Segment(N_, 1);
+  (*new_segment)[0] = polynomials_[dimension];
+  new_segment->setTime(time_);
+  return true;
+}
+
+bool Segment::getSegmentWithAppendedDimension(const Segment& segment_to_append,
+                                              Segment* new_segment) const {
+if (time_ != segment_to_append.getTime()) {
```

Comment:
autoformat, and this comparison sketches me out a bit but then again I think it's actually necessary to avoid numerical problems.